apps/storybook/src/VisCanvasInteraction.stories.tsx

```diff
@@ -18,31 +21,56 @@ interface TemplateProps {
   selection?: 'line' | 'rectangle';
 }
 
+function vectorToStr(vec: Vector2) {
+  return `(${format('.2f')(vec.x)}, ${format('.2f')(vec.y)})`;
+}
+
 const Template: Story<TemplateProps> = (args) => {
   const { panZoom = false, tooltipValue, guides, selection } = args;
 
+  const [selectedVectors, setSelectedVectors] = useState<[Vector2, Vector2]>();
+
   return (
-    <VisCanvas
-      abscissaConfig={{ visDomain: [-10, 0], showGrid: true }}
-      ordinateConfig={{ visDomain: [50, 100], showGrid: true }}
-    >
-      {panZoom && <PanZoomMesh />}
-      {tooltipValue && (
-        <TooltipMesh
-          guides={guides}
-          renderTooltip={(x, y) => (
-            <>
-              {`x=${formatTooltipVal(x)}, y=${formatTooltipVal(y)}`}
-              <div>
-                <strong>{tooltipValue}</strong>
-              </div>
-            </>
-          )}
-        />
+    <>
+      {selection && (
+        <p style={{ textAlign: 'center' }}>
+          {selectedVectors
+            ? `Selection from ${vectorToStr(
+                selectedVectors[0]
+              )} to ${vectorToStr(selectedVectors[1])}`
+            : 'No selection'}
+        </p>
       )}
-      {selection === 'line' && <LineSelectionMesh />}
-      {selection === 'rectangle' && <RectSelectionMesh />}
-    </VisCanvas>
+      <VisCanvas
+        abscissaConfig={{ visDomain: [-10, 0], showGrid: true }}
+        ordinateConfig={{ visDomain: [50, 100], showGrid: true }}
+      >
+        {panZoom && <PanZoomMesh />}
+        {tooltipValue && (
+          <TooltipMesh
+            guides={guides}
+            renderTooltip={(x, y) => (
+              <>
+                {`x=${formatTooltipVal(x)}, y=${formatTooltipVal(y)}`}
+                <div>
+                  <strong>{tooltipValue}</strong>
+                </div>
+              </>
+            )}
+          />
+        )}
+        {selection === 'line' && (
+          <LineSelectionMesh
+            onSelection={(start, end) => setSelectedVectors([start, end])}
+          />
+        )}
+        {selection === 'rectangle' && (
+          <RectSelectionMesh
+            onSelection={(start, end) => setSelectedVectors([start, end])}
+          />
+        )}
+      </VisCanvas>
+    </>
```

Comment:
The template of the story is becoming very "selection-specific". Do you think I should move it to a new file `VisCanvasSelection.stories` ?

Reply:
Yeah, with the local state, it's looking a bit bloated, you're right.